tools/serverpod_cli/test/analyzer/models/stateful_analyzer/model_validation/class_serialize_as_test.dart

```diff
@@ -0,0 +1,158 @@
+  var configWithEnabledSerializeAsJsonbByDefault = GeneratorConfigBuilder()
+      .withEnabledSerializeAsJsonbByDefault()
+      .withEnabledExperimentalFeatures([ExperimentalFeature.serializeAsJsonb]).build();
+
+  group('Given a valid class definition when validating', () {
+    var modelSources = [
+      ModelSourceBuilder().withYaml(
+        '''
+        class: Example
+        table: example
+        fields:
+          tags: List<String>
+        ''',
+      ).build()
+    ];
+
+    var collector = CodeGenerationCollector();
+    var analyzer = StatefulAnalyzer(config, modelSources, onErrorsCollector(collector));
+
+    var definitions = analyzer.validateAll();
+
+    var definition = definitions.first as ModelClassDefinition;
+
+    test('then no errors are collected', () {
+      expect(collector.errors, isEmpty);
+    });
+
+    test('then serializationDataType is set to null.', () {
+      expect(definition.serializationDataType, isNull);
+    });
+  });
+
+  group('Given a valid class definition with serializationDataType set to jsonb when validating', () {
+    var modelSources = [
+      ModelSourceBuilder().withYaml(
+        '''
+        class: Example
+        table: example
+        serializationDataType: jsonb
+        fields:
+          tags: List<String>
+        ''',
+      ).build()
+    ];
+
+    var collector = CodeGenerationCollector();
+    var analyzer = StatefulAnalyzer(config, modelSources, onErrorsCollector(collector));
+
+    var definitions = analyzer.validateAll();
+
+    var definition = definitions.first as ModelClassDefinition;
+
+    test('then no errors are collected', () {
+      expect(collector.errors, isEmpty);
+    });
+
+    test('then serializationDataType is set to jsonb.', () {
+      expect(definition.serializationDataType, SerializationDataType.jsonb);
+    });
+  });
+
+  group('Given a valid class definition with serializationDataType set to json when validating', () {
+    var modelSources = [
+      ModelSourceBuilder().withYaml(
+        '''
+        class: Example
+        table: example
+        serializationDataType: json
+        fields:
+          tags: List<String>
+        ''',
+      ).build()
+    ];
+
+    var collector = CodeGenerationCollector();
+    var analyzer = StatefulAnalyzer(config, modelSources, onErrorsCollector(collector));
+
+    var definitions = analyzer.validateAll();
+
+    var definition = definitions.first as ModelClassDefinition;
+
+    test('then no errors are collected', () {
+      expect(collector.errors, isEmpty);
+    });
+
+    test('then serializationDataType is set to json.', () {
+      expect(definition.serializationDataType, SerializationDataType.json);
+    });
+  });
+
+  group('Given a valid class definition with serializationDataType set and the default serialization for project is set, the field should be defined to serialize to json when validating', () {
```

Comment:
Since we are testing the behavior here - and not the specific combination of json in one and jsonb in the other -, it is good to describe the test as the broader behavior. This applies to other tests as well.

```suggestion
  group('Given a project with default serializationDataType set and a class definition with serializationDataType set to a different value, when validating then field serializationDataType inherits model config', () {
```